ui/app/src/state/reducers/preview.ts

```diff
@@ -227,7 +232,43 @@ const reducer = createReducer<GlobalState['preview']>({
       nextState = { ...nextState, currentUrl: payload.nextUrl };
     }
     return nextState;
-  }
+  },
+  [FETCH_AUDIENCES_PANEL]: (state) => ({
+    ...state,
+    audiencesPanel: {
+      ...state.audiencesPanel,
+      isFetching: true
+    }
+  }),
+  [FETCH_AUDIENCES_PANEL_COMPLETE]: (state, { payload }) => {
+    return {
+      ...state,
+      audiencesPanel: {
+        ...state.audiencesPanel,
+        isFetching: false,
+        error: null,
+        contentType: payload[0],
+        model: payload[1]
+      }
+    }
+  },
+  [FETCH_AUDIENCES_PANEL_FAILED]: (state, { payload }) => ({
+    ...state,
+    audiencesPanel: { ...state.audiencesPanel, error: payload.response, isFetching: true }
+  }),
+  [UPDATE_AUDIENCES_PANEL_PROFILE]: (state, { payload }) => ({
+    ...state,
+    audiencesPanel: {
+      ...state.audiencesPanel,
+      model: {
```

Comment:
Don't think you need to spread the model, do you? Just fully replace? You get the full model from the service, correct?